modeling-cmds/src/shared.rs

```diff
@@ -30,6 +30,16 @@ pub struct LinearTransform {
     /// Defaults to (1, 1, 1) (i.e. the same size as the original).
     #[serde(default = "same_scale")]
     pub scale: Point3d<f64>,
+    /// Rotation axis of the replica.
+    /// Defaults to (0, 0, 1) (i.e. the Z axis).
+    #[serde(default = "z_axis")]
+    pub axis: Point3d<f64>,
+    /// Rotate the replica this far about the rotation axis.
+    /// Defaults to zero (i.e. no rotation).
+    #[serde(default)]
+    pub angle: Angle,
+    /// Origin of the rotation. If one isn't provided, the replica will rotate about its own bounding box center.
+    pub rotation_origin: OriginType,
```

Comment:
I think these 3 properties should be moved into their own `struct Rotation` in `shared.rs`, and this should just be a field `rotation: Rotation`.